### ISSUE TITLE
Problem with Switch/Toggle, default values and DB

### ISSUE THREAD

**Issue by manuelmoreale** [themeum/kirki#879]
### Issue description:

There's a problem with the switch control (and the same problem with the toggle control).
Let's say I have a field like this one

```
Kirki::add_field( 'settings', array(
    'type'     => 'toggle',
    'settings' => 'show-something',
    'default'  => true
))
```

In my case I want to fallback to `true` if the user has not updated the value in the customizer so I have a standard `get_option('show-something' , true)` and everything works as expected.

Now if the user goes in the customizer, changes the value to false and hits save, I should have a new row in the db with a value of false (or empty) but looks like that's not the case.

The row in the db gets added only when the user saves with a value of true. At that point everything works as expected and if the value is then changed back to false the db gets correctly updated.

So to sum up

```
When no row in the DB is present :
From (default) true to (user) false -> Row in DB doesn't get created
From (default) false to (user) true -> Row is created with value of 1 as expected

Once the row is in the db everything works perfectly
```
### Version used:

2.3.0
### Using theme_mods or options?

theme_option

**Comment by aristath** [comment]
I believe this last commit fixed the issue, can you please check using the develop branch?

**Comment by manuelmoreale** [comment]
@aristath sadly the bug is still there.
I [recorded a quick screencast](http://quick.as/q3WS9Waq) that shows a bit better where the problem is.
Let me know if you need more informations or If you want me to test anything.
I'm using the latest version in the dev branch

**Comment by manuelmoreale** [comment]
I add this that's maybe related.
I'm messing with the field trying to work out why is not working correctly and if I dump the content of the values I get something interesting.
On the frontend, with a value of 1 in the DB I get `'visibilityTitle' => string '1' (length=1)` as expected but in the customizer if I switch the toggle to false and then back to true I get `'visibilityLogo' => boolean true`.
Not a big deal since both are `true` values but still interesting nonetheless.

**Comment by manuelmoreale** [comment]
I think we can close this one.
